src/modules/m_filter.cpp

```diff
@@ -646,6 +677,8 @@ bool ModuleFilter::StringToFilterAction(const std::string& str, FilterAction& fa
 		fa = FA_SILENT;
 	else if (stdalgo::string::equalsci(str, "kill"))
 		fa = FA_KILL;
+	else if (stdalgo::string::equalsci(str, "shun") && (ServerInstance->XLines->GetFactory("SHUN") != NULL))
```

Comment:
It might be better to check for the existence of shuns at apply time and write to the log/send a snotice if it can't be found rather than here.

Reply:
I think this is more wide-ranging since this prevents the action from ever becoming FA_SHUN.